packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.spec.ts

```diff
@@ -390,6 +411,55 @@ describe('PayPalCommercePaymentStrategy', () => {
             expect(paymentIntegrationService.submitOrder).toHaveBeenCalled();
         });
 
+        it('triggers restart flow when specified error occurs', async () => {
+            const payload = {
+                payment: {
+                    methodId: defaultMethodId,
+                },
+            };
+
+            const providerError = {
+                status: 'error',
+                three_ds_result: {
+                    acs_url: null,
+                    payer_auth_request: null,
+                    merchant_data: null,
+                    callback_url: null,
+                },
+                errors: [
+                    {
+                        code: 'invalid_request_error',
+                        message:
+                            'Were experiencing difficulty processing your transaction. Please contact us or try again later.',
+                    },
+                    {
+                        code: 'transaction_rejected',
+                        message: 'Payment was declined. Please try again.',
+                        provider_error: {
+                            code: 'INSTRUMENT_DECLINED',
+                        },
+                    },
+                ],
+            };
+
+            jest.spyOn(paypalCommerceIntegrationService, 'submitPayment').mockImplementation(() => {
+                // eslint-disable-next-line @typescript-eslint/no-throw-literal
+                throw providerError;
+            });
+
+            await strategy.initialize(initializationOptions);
+
+            eventEmitter.emit('onApprove');
+            await new Promise((resolve) => process.nextTick(resolve));
+
+            try {
+                await strategy.execute(payload);
+                await new Promise((_resolve, reject) => process.nextTick(reject));
```

Comment:
Do we really need this line of code here?

Reply:
Yes, we need this, because without this line it didn't trigger catch flow for some reason

Reply:
It happens because strategy.execute does not throw an error under the hood. Actually you can mock `submitPayment` method to throw an error, and only then it will work.